runtime-modules/content/src/lib.rs

```diff
@@ -2348,6 +2353,39 @@ decl_module! {
             }
         }
 
+        /// Claims an accumulated channel reward for a council.
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn claim_council_reward(
+            origin,
+            channel_id: T::ChannelId,
+        ) {
+            let channel = Self::ensure_channel_exists(&channel_id)?;
+
+            ensure_actor_authorized_to_claim_council_reward::<T>(origin, &channel.owner)?;
+
+            ensure!(
+                channel.transfer_status == ChannelTransferStatus::NoActiveTransfer,
+                Error::<T>::InvalidChannelTransferStatus
+            );
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            let channel_account_id = ContentTreasury::<T>::account_for_channel(channel_id);
+            let reward: BalanceOf<T> = Balances::<T>::usable_balance(&channel_account_id);
+
+            if !reward.is_zero(){
```

Comment:
Can we do this check, with error if `is_zero`, before we go into mutation safe zone and also trigger an event. I think the simple rule of no side effect => no event is useful for all extrinsics which are supposed to really have a side effect.

Reply:
Sure. Fixed: https://github.com/Joystream/joystream/pull/3562/commits/84782b5ed856271539b530963a5689424e322a03